izer/assets.py

```diff
@@ -222,36 +226,245 @@ def vscode(
             if file.startswith(template_prefix):
 
                 # There is a template file to copy.  Perform string substitution in output file.
-                out_loc = Path(out_path).joinpath(file[len(template_prefix):])  # Remove prefix
+                out_file = Path(out_path).joinpath(file[len(template_prefix):])  # Remove prefix
                 template = Path(directory).joinpath(file)
-                with open(template, 'r', encoding="UTF-8") as in_file, \
-                        open(out_loc, "w+", encoding="UTF-8") as out_file:
-                    for line in in_file.readlines():
-                        out_file.write(
-                            line.replace("##__TARGET__##", target.upper()).
-                            replace("##__BOARD__##", board).
-                            replace("##__PROGRAM_FILE__##", program_file).
-                            replace("##__SYMBOL_FILE__##", symbol_file).
-                            replace("##__M4_OCD_INTERFACE_FILE__##", m4_ocd_interface_file).
-                            replace("##__M4_OCD_TARGET_FILE__##", m4_ocd_target_file).
-                            replace("##__RV_OCD_INTERFACE_FILE__##", rv_ocd_interface_file).
-                            replace("##__RV_OCD_TARGET_FILE__##", rv_ocd_target_file).
-                            replace("\"##__I_PATHS__##\"", i_paths_parsed).
-                            replace("\"##__DEFINES__##\"", defines_parsed).
-                            replace("\"##__V_PATHS__##\"", v_paths_parsed).
-                            replace("##__V_ARM_GCC__##", v_arm_gcc).
-                            replace("##__V_XPACK_GCC__##", v_xpack_gcc).
-                            replace("##__OCD_PATH__##", ocd_path).
-                            replace("##__ARM_GCC_PATH__##", arm_gcc_path).
-                            replace("##__XPACK_GCC_PATH__##", xpack_gcc_path).
-                            replace("##__MAKE_PATH__##", make_path)
-                        )
 
-                os.chmod(out_loc, stat.S_IRWXU | stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH)
-                # print(f"Wrote {os.path.basename(out_loc)}")  # Uncomment to debug
+                content = None
+                with open(template, 'r', encoding="UTF-8") as f:
+                    content = f.read()
+                    content = content.replace("##__TARGET__##", target.upper()). \
+                        replace("##__BOARD__##", board). \
+                        replace("##__PROGRAM_FILE__##", program_file). \
+                        replace("##__SYMBOL_FILE__##", symbol_file). \
+                        replace("##__M4_OCD_INTERFACE_FILE__##", m4_ocd_interface_file). \
```

Comment:
Is there documentation for these? The others are in README.md

Reply:
Yes, see
* [program_file](https://github.com/Analog-Devices-MSDK/VSCode-Maxim/tree/a777071f0309142dc333390fe80fc1d0d9526cfc#program_file)
* [symbol_file](https://github.com/Analog-Devices-MSDK/VSCode-Maxim/tree/a777071f0309142dc333390fe80fc1d0d9526cfc#symbol_file)
* [M4_OCD_interface_file](https://github.com/Analog-Devices-MSDK/VSCode-Maxim/tree/a777071f0309142dc333390fe80fc1d0d9526cfc#m4_ocd_interface_file)